app/src/marketplace/containers/EditProductPage/usePublish.js

```diff
@@ -125,8 +130,33 @@ export default function usePublish() {
             }
         }
 
-        // update price, currency & beneficiary if changed. This will also
-        // do republish for products that have been at some point deployed
+        // update whitelist enabled status if it has changed
+        if ([publishModes.REPUBLISH, publishModes.REDEPLOY].includes(nextMode)) {
+            if (hasRequireWhitelistChanged && contractProduct) {
```

Comment:
Does this need its own transaction? Or is the product update enough below

Reply:
Looks like not ☹️ That could have been added as part of contract update, might be too late now.

Reply:
Yep. That's how we roll for now.